tests/masking_test.py

```diff
@@ -325,10 +138,102 @@ def padded_sum(x):
     expected = np.array([0, 1, 2, 3, 4])
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def check(self, fun, input_shapes, values_dict,
+            out_shape=None, unpadded_vars=None, custom_inputs=None,
+            skip_shapecheck=False, check_output_fun=None):
```

Comment:
This helper function looks really bloated. Doesn't it make sense to have separate tests for masking and shape checking? I think having more, fine-grained tests is probably better than few tests where each test checks two different things.

I'm also not sure whether we need to check vmap(mask) for every primitive, since by that logic you should probably test transformation(mask) for every other transformation. I think it makes sense to test vmap(mask) for one or two simple primitives, and any other primitives that we suspect might be problematic. What do you think?

My suggestion would be to revert to the style we had on master, but feel free to disagree and explain why you think this is better. If nothing else it makes the diff a lot easier to review 😜.

Reply:
On call @JuliusKunze has convinced me that we should test vmap(mask) for each primitive, however it might still be worth trying to simplify this function.